### ISSUE TITLE
Add support for comments for matrix questions

### ISSUE THREAD

**Issue by gjvnq** [surveyjs/survey-library#3396]
### Are you requesting a feature, reporting a bug or asking a question?

Requesting a feature, probably.

### What is the current behavior?

The option to add comments to a question (`hasComment` and  `commentText`) don't work for matrix questions.

The option does not appear in the UI and trying to "code it in" produces no effect.

### What is the expected behavior?

Probably the one that is happening. But I wish the comments field would appear to both matrix and non-matrix questions.

### How would you reproduce the current behavior (if this is a bug)?

Create a new survey with Survey Creator and use the JSON provided below.

```json
{
 "pages": [
  {
   "name": "page1",
   "elements": [
    {
     "type": "radiogroup",
     "name": "question1",
     "hasComment": true,
     "commentText": "Explain you answer, please",
     "choices": [
      "item1",
      "item2",
      "item3"
     ]
    },
    {
     "type": "matrix",
     "name": "question2",
     "hasComment": true,
     "commentText": "Explain you answer, please",
     "columns": [
      "Column 1",
      "Column 2",
      "Column 3"
     ],
     "rows": [
      "Row 1",
      "Row 2"
     ]
    }
   ]
  }
 ]
}
```

### Specify your

* browser: Chrome
* browser version: 93.0.4577.63
* surveyjs platform (angular or react or jquery or knockout or vue): https://surveyjs.io/create-survey
* surveyjs version: aparently  v1.8.68

**Comment by andrewtelnov** [comment]
@gjvnq I am not sure that we want to change it by default.
We can add a new property nto Survey settings.
If you run Survey Creator on your own website, then it can set Survey settings in the code. Will it work for you?

Thank you,
Andrew

**Comment by gjvnq** [comment]
> If you run Survey Creator on your own website, then it can set Survey settings in the code. Will it work for you?

I don't understand what you are trying to say.

**Comment by gjvnq** [comment]
Instead of things being like this:

![Screenshot from 2021-09-29 23-27-55](https://user-images.githubusercontent.com/3050524/135375786-167f22cb-b662-4cd1-aa93-c7899fceedc7.png)

I wish they could be like this:

![Screenshot from 2021-09-29 23-26-46](https://user-images.githubusercontent.com/3050524/135375698-64085753-f826-416a-b31a-da375f7a597c.png)

Notice the comments field at the bottom of the matrix question.

**Comment by andrewtelnov** [comment]
@gjvnq Are you talking about Survey Creator on our site, am I correct?

Thank you,
Andrew

**Comment by gjvnq** [comment]
> Are you talking about Survey Creator on our site, am I correct?

Yes. But I'm talking not about the default settings, but about what features are currently implemented.

Notice how there is an option for a comment for non-matrix questions:

![Screenshot from 2021-09-29 23-24-56](https://user-images.githubusercontent.com/3050524/135455999-a5d2c2ae-ffe7-47ef-b00a-7bb3ee45cc1f.png)

But there isn't such an option for a matrix question:

![Screenshot from 2021-09-29 23-25-05](https://user-images.githubusercontent.com/3050524/135456065-af190fde-2de8-4b47-b438-ff14491d74ca.png)

**Comment by andrewtelnov** [comment]
@gjvnq We have hide this functionality for some questions. It was several years ago. I do not remember why we did it exactly.
We have to discuss it.

Thank you,
Andrew

**Comment by gjvnq** [comment]
I don't think that this feature is merely hidden but disabled or, more likely to me, unimplemented.

I tried editing the JSON manually (see the code/config I wrote on the first comment in this issue) and it produced no effect.

**Comment by gjvnq** [comment]
I looked a bit into the source code and it really seems like an unimplemented feature.

The files `src/question_matrixdropdownbase.ts` and `src/question_matrixdropdownbase.ts` suggest that the comments field is used to store what the user wrote in the "other" option.

However, no similar feature appears to be implemented for `src/question_matrix.ts` (an its imports `src/base.ts` and `src/martixBase.ts` and `src/question.ts`).

**Comment by andrewtelnov** [comment]
@gjvnq I wrote this code :) To make it available in all questions I have to change one line, one word. [Here it is](https://github.com/surveyjs/survey-library/blob/master/src/question.ts#L893).

Thank you,
Andrew

**Comment by gjvnq** [comment]
> I wrote this code :)

Ooops. I thought the project was too big for a single person to write it all and answer the issues tickets.

**Comment by andrewtelnov** [comment]
@gjvnq  I mean this part, related to comments :)

**Comment by benmoses-equineregister** [comment]
**Note:** The change described above on it's own didn't work for me, I also had to copy the 3 properties `hasComment`, `commentText` and `commentPlaceHolder` from `question_baselect.ts` to the question class so all questions had them (I need them for all questions).

**Comment by JaneSjs** [comment]
Hi @benmoses-equineregister,
Please be informed that all SurveyJS question types support an additional comment area. Please upgrade your SurveyJS Form Library to the latest [v2.0.10](https://surveyjs.io/stay-updated/release-notes/v2.0.10) and run the following survey JSON:
```
{
  "pages": [
    {
      "name": "page1",
      "elements": [
        {
          "type": "radiogroup",
          "name": "question1",
          "showCommentArea": true,
          "commentText": "Explain you answer, please",
          "choices": [
            "item1",
            "item2",
            "item3"
          ]
        },
        {
          "type": "matrix",
          "name": "question2",
          "showCommentArea": true,
          "commentText": "Explain you answer, please",
          "columns": [
            "Column 1",
            "Column 2",
            "Column 3"
          ],
          "rows": [
            "Row 1",
            "Row 2"
          ]
        }
      ]
    }
  ]
}
```
Comment sections are available for all question types:

![Image](https://github.com/user-attachments/assets/24847d23-2d07-4dae-8d0e-c46f3a207c76)

Should you have any further questions, please feel free to contact us at any time.
